src/language.cpp

```diff
@@ -310,14 +339,46 @@ std::string Language::GetText(StringID number)
 	if (number < lengthof(default_strings)) return default_strings[number];
 
 	if (number < lengthof(this->registered) && this->registered[number] != nullptr) {
-		const std::string &text = this->registered[number]->GetString();
+		const std::string &text = this->registered[number]->GetString(amount);
 		if (text.empty()) return "<empty text>";
 		return text;
 	}
 
 	return "<Invalid string>";
 }
 
+/**
+ * Get the singular form for string number \a number.
+ * @param number string number to get.
+ * @return String corresponding to the number (not owned by the caller, so don't free it).
+ */
+std::string Language::GetSgText(StringID number)
+{
+	return this->GetPlural(number, 1);
+}
+
+/**
+ * Look up the correct plural form for an amount.
+ * @param lang_index The language to look in.
+ * @param amount The amount to look up.
+ * @return Index of the correct plural form.
+ */
+uint Language::GetPluralFormIndex(int lang_index, int64 amount)
+{
+	if (amount == 1 && this->singular_form_index[lang_index] != -1) return this->singular_form_index[lang_index];
+
+	int result = this->plural_forms[lang_index]->Eval(amount);
```

Comment:
I am not sure why you cache(?) singular values.

Is "Eval" that expensive?

It looks like something that shouldn't be needed.